### PR TITLE
Add MSpec integration

### PR DESCRIPTION
**Pull request by Slesa**
The new dependency is 
```
<PackageReference Include="Machine.Specifications.Runner.Console" version="0.9.3" />
```

It calls mspec-clr4.exe

### CODE REVIEW

File: build/specifications/MSpec.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    {
+      "settingsClass": {
+        "properties": [
+          {
+            "name": "Assemblies",
+            "type": "List<string>",
+            "format": "{value}",
+            "separator": " ",
+            "createOverload": true,
+            "help": "Assemblies with tests to be executed executed."
```
Comment (arodus) [inline build/specifications/MSpec.json]: Duplicate executed

File: build/specifications/MSpec.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            "format": "-f={value}",
+            "separator": ",",
+            "help": "Filter file specifying contexts to execute (full type name, one per line). Takes precedence over tags."
+          },
+          {
+            "name": "Includes",
+            "type": "List<string>",
+            "format": "-i={value}",
+            "separator": ",",
+            "help": "Executes all specifications in contexts with these comma delimited tags. Ex. -i 'foo, bar, foo_bar'."
```
Comment (arodus) [inline build/specifications/MSpec.json]: Use `<c>...</c>` for examples.

File: build/specifications/MSpec.json [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            "type": "List<string>",
+            "format": "-x={value}",
+            "separator": ",",
+            "help": "Exclude specifications in contexts with these comma delimited tags. Ex. -x 'foo, bar, foo_bar'."
+          },
+          {
+            "name": "HtmlOutput",
+            "type": "string",
+            "format": "--html={value}",
+            "help": "Outputs the HTML report to path, one-per-assembly w/ index.html (if directory, otherwise all are in one file)."
```
Comment (arodus) [inline build/specifications/MSpec.json]: Use `<c>...</c>` for examples.

File: build/specifications/MSpec.json [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            "format": "-i={value}",
+            "separator": ",",
+            "help": "Executes all specifications in contexts with these comma delimited tags. Ex. -i 'foo, bar, foo_bar'."
+          },
+          {
+            "name": "Excludes",
+            "type": "List<string>",
+            "format": "-x={value}",
+            "separator": ",",
+            "help": "Exclude specifications in contexts with these comma delimited tags. Ex. -x 'foo, bar, foo_bar'."
```
Comment (arodus) [inline build/specifications/MSpec.json]: Use `<c>...</c>` for examples.

File: build/specifications/MSpec.json [hunk 5]
```diff
@@ -0,0 +1,108 @@
+{
+  "$schema": "https://raw.githubusercontent.com/nuke-build/nuke/master/source/Nuke.CodeGeneration/schema.json",
+  "references": [
+    "https://github.com/machine/machine.specifications.runner.console"
```
Comment (arodus) [inline build/specifications/MSpec.json]: Use raw url. `https://raw.githubusercontent.com/machine/machine.specifications.runner.console/master/README.md`

**Review by arodus — CHANGES_REQUESTED** [review]

File: build/specifications/MSpec.json [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  ],
+  "name": "Mspec",
```
Comment (arodus) [inline build/specifications/MSpec.json]: `MSpec` instead of `Mspec`.

File: build/specifications/MSpec.json (hunk 1 above)
Comment (Slesa) [reply build/specifications/MSpec.json]: Can you explain, please?

Comment (matkoch) [reply build/specifications/MSpec.json]: Assemblies with tests to be executed executed

File: source/Nuke.Common/Tools/MSpec/MSpecTasks.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿// Copyright 2018 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using Nuke.Common.Tooling;
+
+namespace Nuke.Common.Tools.MSpec
+{
+    partial class MSpecSettings {
```
Comment (arodus) [inline source/Nuke.Common/Tools/MSpec/MSpecTasks.cs]: Curly bracket has to be on a new line.

File: source/Nuke.Common/Tools/MSpec/MSpecTasks.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        private string GetToolPath() {
```
Comment (arodus) [inline source/Nuke.Common/Tools/MSpec/MSpecTasks.cs]: Curly bracket has to be on a new line.

File: source/Nuke.Common/Tools/MSpec/MSpecTasks.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+            return MSpecTasks.GetToolPath();
+        }
+    }
+
+
```
Comment (arodus) [inline source/Nuke.Common/Tools/MSpec/MSpecTasks.cs]: Remove duplicate newline.

File: source/Nuke.Common/Tools/MSpec/MSpec.Generated.cs [hunk 10]
```diff
@@ -0,0 +1,733 @@
...
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Nuke.Common.Tools.Mspec
+{
+    [PublicAPI]
+    [ExcludeFromCodeCoverage]
+    public static partial class MspecTasks
```
Comment (arodus) [inline source/Nuke.Common/Tools/MSpec/MSpec.Generated.cs]: Still called `Mspec`, please generate the tool again.

File: source/Nuke.Common/Tools/MSpec/MSpecTasks.cs (hunk 8 above)
Comment (arodus) [inline source/Nuke.Common/Tools/MSpec/MSpecTasks.cs]: Unless you set `customExecutable=true` in the specification this function is not used.

**Review by arodus — CHANGES_REQUESTED** [review]

File: source/Nuke.Common/Tools/MSpec/MSpecTasks.cs (hunk 7 above)
Comment (arodus) [reply source/Nuke.Common/Tools/MSpec/MSpecTasks.cs]: Also the `public` accessibility  modifier is missing.

**Review by arodus — APPROVED** [review]

**Pull request merged**